src/shared/tests.h

```diff
@@ -227,6 +227,16 @@ static inline int run_test_table(void) {
                 }                                                                                               \
         })
 
+#define ASSERT_FAIL(expr)                                                                                       \
```

Comment:
Why not just use ASSERT_ERROR? We usually (if not always) want to check the errno too?

Reply:
@YHNdnzj We have a lot of tests that don't check the error. I want to be able to migrate those to use the new macros without having to go look up the possible errors for each one so we need a macro that doesn't take a specific error.